modules/SLAPEnrich.py

```diff
@@ -1,287 +1,117 @@
-                                'disease_id': evidenceString.disease.id
-                            }
-
-                            error = evidenceString.validate(logging)
-
-                            if error > 0:
-                                self.logger.error(evidenceString.to_JSON())
-                                sys.exit(1)
-
-                            self.evidence_strings.append(evidenceString)
-
-                        else:
-                            self.logger.error("%s is not found in Ensembl" % gene_symbol)
-
-            self.logger.info("%s evidence parsed"%(n-1))
-            self.logger.info("%s evidence created"%len(self.evidence_strings))
-
-        slapenrich_input.close()
-
-    def write_evidence(self, filename=Config.SLAPENRICH_EVIDENCE_FILENAME):
-        self.logger.info("Writing SLAPEnrich evidence strings")
-        with open(filename, 'w') as slapenrich_output:
-            n = 0
-            for evidence_string in self.evidence_strings:
-                n += 1
-                self.logger.info(evidence_string.disease.id[0])
-                # get max_phase_for_all_diseases
-                error = evidence_string.validate(logging)
-                if error == 0:
-                    slapenrich_output.write(evidence_string.to_JSON(indentation=None)+"\n")
-                else:
-                    self.logger.error("REPORTING ERROR %i" %n)
-                    self.logger.error(evidence_string.to_JSON(indentation=4))
-            slapenrich_output.close()
 
+        return self.dataframe
+
+    @staticmethod
+    def parseEvidenceString(row):
+        try:
+            evidence = {
+                "datasourceId" : "slapenrich",
+                "datatypeId" : "affected_pathway",
+                "diseaseFromSource" : row["Cancer_type_acronym"],
+                "diseaseFromSourceMappedId" if row["EFO_id"] else None : row["EFO_id"],
+                "resourceScore" : row["pval"],
+                "pathwayName" : row["pathwayDescription"],
+                "pathwayId" : row["pathwayId"],
+                "targetFromSourceId" : row["gene"]
+            }
+            evidence.pop(None, None)
+            return evidence
+        except Exception as e:
+            raise        
 
 def main():
-    slap = SLAPEnrich()
-    slap.process_slapenrich()
-
-if __name__ == "__main__":
-    main()
-
+    # Initiating parser
+    parser = argparse.ArgumentParser(description=
+    "This script generates evidences for the SLAPEnrich data source.")
+
+    parser.add_argument("-i", "--inputFile", required=True, type=str, help="Input source .tsv file.")
+    parser.add_argument("-o", "--outputFile", required=True, type=str, help="Name of the evidence compressed JSON file containing the evidence strings.")
+    parser.add_argument("-m", "--mappingStep", required=False, type=bool, default=True, help="State whether to run the disease to EFO term mapping step.")
```

Comment:
I think it doesn't work. In this format, `-m` expects an argument, however, you only use it as a flag. 

Write something like this:

```python
parser.add_argument('-m', action='store_false') # if the flag is present, it's value will be false
```

Reply:
Also if it made the modification to be able to run the script without mapping, the script fails.